docs/table-upgrade.md

```diff
@@ -0,0 +1,41 @@
+# Table Upgrade logic and data structures
```

Comment:
what about GRANTS migration?

Reply:
+1

Reply:
If Grants are used, we should migrate them.

Reply:
https://github.com/databrickslabs/ucx/issues/316 tracking